examples/generative/molecule_generation.py

```diff
@@ -0,0 +1,583 @@
+    features_transformed = features
+    for units in gconv_units:
+        features_transformed = RelationalGraphConvLayer(units)(
+            [adjacency, features_transformed]
+        )
+    # Reduce 2-D representation of molecule to 1-D
+    x = keras.layers.GlobalAveragePooling1D()(features_transformed)
+
+    # Propagate through one or more densely connected layers
+    for units in dense_units:
+        x = layers.Dense(units, activation="relu")(x)
+        x = layers.Dropout(dropout_rate)(x)
+
+    z_mean = layers.Dense(latent_dim, dtype="float32", name="z_mean")(x)
+    log_var = layers.Dense(latent_dim, dtype="float32", name="log_var")(x)
+
+    encoder = keras.Model([adjacency, features], [z_mean, log_var], name="encoder")
+
+    return encoder
+
+
+def get_decoder(dense_units, dropout_rate, latent_dim, adjacency_shape, feature_shape):
+    latent_inputs = keras.Input(shape=(latent_dim,))
+
+    x = latent_inputs
+    for units in dense_units:
+        x = keras.layers.Dense(units, activation="tanh")(x)
+        x = keras.layers.Dropout(dropout_rate)(x)
+
+    # Map outputs of previous layer (x) to [continuous] adjacency tensors (x_adjacency)
+    x_adjacency = keras.layers.Dense(tf.math.reduce_prod(adjacency_shape))(x)
+    x_adjacency = keras.layers.Reshape(adjacency_shape)(x_adjacency)
+    # Symmetrify tensors in the last two dimensions
+    x_adjacency = (x_adjacency + tf.transpose(x_adjacency, (0, 1, 3, 2))) / 2
+    x_adjacency = keras.layers.Softmax(axis=1)(x_adjacency)
+
+    # Map outputs of previous layer (x) to [continuous] feature tensors (x_features)
+    x_features = keras.layers.Dense(tf.math.reduce_prod(feature_shape))(x)
+    x_features = keras.layers.Reshape(feature_shape)(x_features)
+    x_features = keras.layers.Softmax(axis=2)(x_features)
+
+    decoder = keras.Model(
+        latent_inputs, outputs=[x_adjacency, x_features], name="decoder"
+    )
+
+    return decoder
+
+
+"""
+## Build Sampling layer
+"""
+
+
+class Sampling(layers.Layer):
+    def call(self, inputs):
+        z_mean, z_log_var = inputs
+        batch = tf.shape(z_log_var)[0]
+        dim = tf.shape(z_log_var)[1]
+        epsilon = tf.keras.backend.random_normal(shape=(batch, dim))
+        return z_mean + tf.exp(0.5 * z_log_var) * epsilon
+
+
+"""
+## Building the VAE
+
+With this model we try to optimize four losses:
+* Categorical Crossentropy
+* KL Loss
+* Property prediction Loss
+* Graph Loss( Gradient Penalty )
+
+The Categorical Crossentropy loss function is for giving a measure of the model's
+reconstruction capacity. The Property prediction loss is to estimate the mean squared
+error of the predicted and actual property after passing the `z_mean(generated from the
+encoder)` through a property prediction model or property prediction layer. The property
+prediction of the model is optimized through binary cross-entropy and the gradient
+penalty is further guided by the model's property(QED) prediction.
```

Comment:
The last sentence is unclear, please rephrase. What is the gradient penalty for?